IA-1613 expose the creator of an org unit in the api and show it in the org unit details

SimonDus opened the pull request:
- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Did I add translations
## Changes

Expose OU creator in OU List and ou Details


## How to test

Create an org unit, go to ou List or Ou details. 

## Print screen / video

![image](https://user-images.githubusercontent.com/45785235/200255963-75b3fd73-4c09-4307-bad6-831b480321a6.png)

![image](https://user-images.githubusercontent.com/45785235/200256068-7d492088-1365-4ef4-92e4-0aeda52383d0.png)

Commit:
add creator to OU response

Commit:
add OU creator to backend and fronted

Commit:
add to details

Commit:
black format

beygorghor reviewed (approved):
- "en" translation was missing for "iaso.orgUnits.createdBy"
- i removed width on id and group cell to make the columns fit automatically
- Also display a placeholder instead of nothing if no creator
I did those changes ;-)

beygorghor committed:
small fixes

"en" translation was missing for "iaso.orgUnits.createdBy"
i removed width on id and group cell to make the columns fit automatically
Also display a placeholder instead of nothing if no creator
I did those changes ;-)

niconoe commented on iaso/api/serializers.py:
```diff
@@ -104,10 +107,14 @@ class Meta:
 class OrgUnitSmallSearchSerializer(OrgUnitSerializer):
+    def get_creator(self, org_unit):
+        return None if org_unit.creator is None else org_unit.creator.username
+
```
Is it necessary to redefine here if the same code already appears in the parent class?

SimonDus replied:
Indeed, I forgot to remove it as the "OrgUnitSmallSearchSerializer" is used for the details and I did it first.

niconoe commented on iaso/api/serializers.py:
```diff
@@ -143,6 +152,15 @@ def get_instances_count(self, org_unit):
         else:
             return org_unit.instance_set.filter(~Q(file="") & ~Q(device__test_device=True) & ~Q(deleted=True)).count()
 
+    def get_creator(self, org_unit):
```
I'm not familiar with the code, but I'm not sure why we redefine here an improved(?) version of get_creator() here. Would the improved version work everywhere (in that case I suggest moving it to the parent class and letting others inherit it?). Or is there a good reason for a different behavior in `OrgUnitSearchSerializer`? If so, can we add a comment explaining the "why"? thanks!

niconoe commented on iaso/tests/api/test_serializers.py:
```diff
@@ -251,10 +254,12 @@ def test_small_search(self):
                 "name": "Endor Jedi Squad 1",
                 "org_unit_type_name": "Jedi Squad",
                 "parent": {
+                    "creator": None,
```
Maybe we can improve the tests (or add a new one), to make sure the creator field has the expected value in case it's not None? thanks!

SimonDus replied:
Yep, done

niconoe reviewed (changes requested):
Thanks a lot for the good work, I just let a few small comments.

Commit:
remove unnecessary code + add test

Commit:
Merge branch 'IA-1613-Expose-the-creator-of-an-org-unit-in-the-api-and-show-it-in-the-org-unit-details' of github.com:BLSQ/iaso into IA-1613-Expose-the-creator-of-an-org-unit-in-the-api-and-show-it-in-the-org-unit-details

Commit:
black formatting

olethanh commented:
I would say it need a prefetch_related to not create to much query. But I'm not sure which endpoint get impacted by this change. 

Also not sure why we are returning a string and not an embeded object like we do in other parts ?

SimonDus commented:
> I would say it need a prefetch_related to not create to much query. But I'm not sure which endpoint get impacted by this change.
> 
> Also not sure why we are returning a string and not an embeded object like we do in other parts ?



> I would say it need a prefetch_related to not create to much query. But I'm not sure which endpoint get impacted by this change.
> 
> Also not sure why we are returning a string and not an embeded object like we do in other parts ?

Because Son told me that on other part of Iaso we use " Username ( first_name last_name ) " I think in budget. So I made something similar. But for me it can return only the username... It's kinda the same to me

olethanh commented:
I think he mean the function getDisplayName in userUtils.ts but that  is in the frontend. I'm merging we can do the rest later
